pkg/providers/vsphere/vmprovider_vm_resize_test.go

```diff
@@ -156,25 +176,97 @@ func vmResizeTests() {
 			It("Resizes", func() {
 				cs := configSpec
 				cs.MemoryMB = 8192
-				vm.Spec.ClassName = createVMClass(cs).Name
+				newVMClass := createVMClass(cs)
+				vm.Spec.ClassName = newVMClass.Name
 
 				vcVM, err := createOrUpdateAndGetVcVM(ctx, vm)
 				Expect(err).ToNot(HaveOccurred())
 
 				var o mo.VirtualMachine
 				Expect(vcVM.Properties(ctx, vcVM.Reference(), nil, &o)).To(Succeed())
 				Expect(o.Config.Hardware.MemoryMB).To(BeEquivalentTo(8192))
+
+				assertExpectedLastResizeAnnotation(vm, newVMClass)
 			})
 		})
 
-		Context("Devops Overrides", func() {
+		Context("Same Class Resize Annotation", func() {
+			BeforeEach(func() {
+				configSpec.MemoryMB = 1024
+			})
 
+			It("Resizes", func() {
+				cs := configSpec
+				cs.MemoryMB = 8192
+				updateVMClass(vmClass, cs)
+
+				By("Does not resize without annotation", func() {
+					vcVM, err := createOrUpdateAndGetVcVM(ctx, vm)
+					Expect(err).ToNot(HaveOccurred())
+
+					var o mo.VirtualMachine
+					Expect(vcVM.Properties(ctx, vcVM.Reference(), nil, &o)).To(Succeed())
+					Expect(o.Config.Hardware.MemoryMB).To(BeEquivalentTo(1024))
+				})
+
+				vm.Annotations[vmopv1.VirtualMachineSameVMClassResizeAnnotation] = ""
+				vcVM, err := createOrUpdateAndGetVcVM(ctx, vm)
+				Expect(err).ToNot(HaveOccurred())
+
+				var o mo.VirtualMachine
+				Expect(vcVM.Properties(ctx, vcVM.Reference(), nil, &o)).To(Succeed())
+				Expect(o.Config.Hardware.MemoryMB).To(BeEquivalentTo(8192))
+
+				assertExpectedLastResizeAnnotation(vm, vmClass)
+			})
+		})
+
+		Context("Devops Overrides", func() {
 			Context("ChangeBlockTracking", func() {
+				It("Overrides", func() {
+					vm.Spec.Advanced = &vmopv1.VirtualMachineAdvancedSpec{
+						ChangeBlockTracking: vimtypes.NewBool(true),
+					}
+
+					vcVM, err := createOrUpdateAndGetVcVM(ctx, vm)
+					Expect(err).ToNot(HaveOccurred())
+
+					var o mo.VirtualMachine
+					Expect(vcVM.Properties(ctx, vcVM.Reference(), nil, &o)).To(Succeed())
+					Expect(o.Config.ChangeTrackingEnabled).To(HaveValue(BeTrue()))
+
+					assertExpectedLastResizeAnnotation(vm, vmClass)
+				})
+			})
+
+			Context("VM Class does not exist", func() {
 				BeforeEach(func() {
 					configSpec.ChangeTrackingEnabled = vimtypes.NewBool(false)
 				})
 
-				It("Overrides", func() {
+				It("Still applies overrides", func() {
+					Expect(ctx.Client.Delete(ctx, vmClass)).To(Succeed())
+
+					vm.Spec.Advanced = &vmopv1.VirtualMachineAdvancedSpec{
+						ChangeBlockTracking: vimtypes.NewBool(true),
+					}
+
+					vcVM, err := createOrUpdateAndGetVcVM(ctx, vm)
+					Expect(err).ToNot(HaveOccurred())
+
+					var o mo.VirtualMachine
+					Expect(vcVM.Properties(ctx, vcVM.Reference(), nil, &o)).To(Succeed())
+					Expect(o.Config.ChangeTrackingEnabled).To(HaveValue(BeTrue()))
+				})
+			})
+
+			Context("VM Classless VMs", func() {
+				BeforeEach(func() {
+					configSpec.ChangeTrackingEnabled = vimtypes.NewBool(false)
```

Comment:
Nit: could we let the default val (false) just be picked up here instead of setting it explicitly?. thereby simulating the classlessVM scenario.

Reply:
It kind of doesn't matter. What makes the VM classless is when we set the `Spec.ClassName = ""`. The CS value here is used to create the VM so either we can test that it goes from `nil -> &true` or`&false -> &true`, and I kind of prefer the later.

Reply:
> What makes the VM classless is when we set the Spec.ClassName = ""
agree.
> The CS value here is used to create the VM so either we can test that it goes from nil -> &true or&false -> &true, and I kind of prefer the later.
I wanted us to have testing as close to the actual implementation. As in, What would the overwrite do to a Classless VM's ConfigSpec ( which would only have default values). In testing, we could test by manually setting
`configSpec.ChangeTrackingEnabled = vimtypes.NewBool(false) ` but that's not what would happen actually with the happy path for classless. 

Anyways, it's just a Nit. I do see the benefit of having it set like that in BeforeEach().

Reply:
I'm still a little confused: for classless we won't have a class ConfigSpec so it will be default values. The BeforeEach is just used for creating the VM. This test is faking how real classless VMs would be "created" since they are really imported. But it is possible for an classless/imported VM to have this field - or any other - set.

Reply:
I see what you mean. 
I was under the assumption that we were setting value (ChangeBlockTracking:false) for the configSpec argument in [OverwriteResizeConfigSpec() ](https://github.com/vmware-tanzu/vm-operator/blob/ef715a3111264925ab92ba06f66ac390430641cc/pkg/providers/vsphere/session/session_vm_update.go#L881)
```
if err := vmopv1util.OverwriteResizeConfigSpec(vmCtx, *vmCtx.VM, *moVM.Config, &configSpec); err != nil {
		return false, err
	}
```

But I see why you said it 'doesn't matter' now since the value we set is only used to create a vmclass() in the test which does really get used in the ClasslessVM scenario because we will not enter [this block](https://github.com/vmware-tanzu/vm-operator/blob/ef715a3111264925ab92ba06f66ac390430641cc/pkg/providers/vsphere/session/session_vm_update.go#L869) below and use this configSpec. 

`        if resizeArgs.VMClass != nil {`